testdata/sqllogictests/functions/cosine_similarity.slt

```diff
@@ -0,0 +1,45 @@
+# tests for cosine similarity function
```

Comment:
I think we should probably also have some small collection of tests  where the input numbers (to the table, the vector, the target) that are negative, have non-zero decimal point values, and are less than one, and numbers that would overflow a single precision float.

Reply:
I think those would probably be best tested inside lance directly. We are just a thin wrapper over lance's implementation, so if there are issues like the ones you mention, I'd expect that those should likely be opened and handled upstream.

Reply:
Yeah im going to have to fundamentally disagree here. We shouldn't be testing the function itself, and it's generally a bad practice to essentially unit test external functions. 

We test the **integration**, if the function has bugs such as handling of negative values, then that should be taken upstream. We add a few tests like the ones here for testing the integration with our own system. We want to test **OUR** logic, generally we assume that external libraries code is well tested and if bugs arise we open it up with them, not test it internally.

--- 
Ill add some tests just to get this approved, but I **strongly** disagree with the decision to add them as i think it is not good practice to test external functions like this.